myapp/app/controllers/tasks_controller.rb

```diff
@@ -26,19 +26,21 @@ def index
     # TODO: support ascending
     sort += ' DESC'
 
-    @tasks = q.order(sort).page(params[:page])
+    @tasks = q.where(user_id: current_user.id).order(sort).page(params[:page])
 
     # for new form
     @new_task = Task.new
   end
 
   def show
     @task = Task.find_by(id: params[:id])
-    redirect_to error_path(404) if @task.nil?
+    return redirect_to error_path(404) if @task.nil?
+    return redirect_to error_path(401) if @task["user_id"] != current_user.id
   end
 
   def create
     data = task_data(params)
+    data["user_id"] = current_user.id
```

Comment:
[FYI] Using strong parameters is a more Rails-like way.
https://railsguides.jp/action_controller_overview.html#strong-parameters

```ruby
    @task = current_user.tasks.build(task_data)
```


```ruby
  def task_data
    params.require(:task).permit(:title, :description, :due_date_at, :status)
  end
```

Reply:
Thanks for the suggestion. I updated and differenciate the params for #create and #update as below.

```
  def create_params
    # value coming from select field is string, convert it to int explcitly here
    params[:task][:status] = params[:task][:status].to_i
    params[:task][:user_id] = current_user.id
    params.require(:task).permit(:title, :description, :due_date_at, :status, :user_id)
  end

  def update_params
    # value coming from select field is string, convert it to int explcitly here
    params[:task][:status] = params[:task][:status].to_i
    params.require(:task).permit(:title, :description, :due_date_at, :status)
  end
```